### PR TITLE
very big, ginormous reorganization

### PR DESCRIPTION
**Pull request by saucepoint**
## Related Issue
Reorganizes posm

## Description of changes

* Move state, permission modifiers, and ERC721Permit to `PositionActionsRouter`
* `PositionActionsRouter` inherits Notifier
* Move external Notifier functions to `Notifier.sol`
* Cleaned tf out of the imports
* Defined interfaces for every subcontract, **still missing natspec**

image

### CODE REVIEW

File: src/PositionManager.sol [hunk 1]
```diff
@@ -2,91 +2,16 @@
...
-import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
-import {BalanceDelta} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
-import {SafeCast} from "@uniswap/v4-core/src/libraries/SafeCast.sol";
-import {Position} from "@uniswap/v4-core/src/libraries/Position.sol";
-import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
-import {TransientStateLibrary} from "@uniswap/v4-core/src/libraries/TransientStateLibrary.sol";
-import {Position} from "@uniswap/v4-core/src/libraries/Position.sol";
-import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
-import {ERC20} from "solmate/src/tokens/ERC20.sol";
 import {IAllowanceTransfer} from "permit2/src/interfaces/IAllowanceTransfer.sol";
```
Comment (saucepoint) [inline src/PositionManager.sol:5]: cleannnn

File: src/PositionManager.sol [hunk 2]
```diff
@@ -2,91 +2,16 @@
...
+import {Multicall_v4} from "./base/Multicall_v4.sol";
 
-    /// @notice Reverts if the hash of the config does not equal the saved hash
-    /// @param tokenId the unique identifier of the ERC721 token
-    /// @param config the PositionConfig to check against
-    modifier onlyValidConfig(uint256 tokenId, PositionConfig calldata config) {
-        if (positionConfigs.getConfigId(tokenId) != config.toId()) revert IncorrectPositionConfigForTokenId(tokenId);
-        _;
-    }
+contract PositionManager is IPositionManager, PositionActionsRouter, PoolInitializer, Permit2Forwarder, Multicall_v4 {
```
Comment (saucepoint) [inline src/PositionManager.sol:13]: too clean 🧐 🥵

Comment (snreynolds) [reply src/PositionManager.sol:13]: i like

**Comment by saucepoint** [comment]
we went too crazy on this mf so replaced by https://github.com/Uniswap/v4-periphery/pull/287

**Pull request closed without merging**